### ISSUE TITLE
Decide how we should deal with math/special_functions/chebyshev_transform.hpp

### ISSUE THREAD

**Issue by cosurgi** [BoostGSoC21/math#20]
The file [chebyshev_transform.hpp](https://github.com/BoostGSoC21/math/blob/develop/include/boost/math/special_functions/chebyshev_transform.hpp#L10) uses FFT. We should know how to deal with this.

**Comment by cosurgi** [comment]
Ignoring this is also one of possible solutions. Usually these kinds of architectural problems led to a better design overall, and required drastic redesign along the way.

**Comment by ckormanyos** [comment]
>  file `chebyshev_transform.hpp` uses FFT. We should know how to deal with this.

I had noticed that as well.

> Ignoring this is also one of possible solutions. 

My strategy had been (and continues to be) along the lines of _ignoring_, but not entirely and forever. Rather, igonre until we have a specified and established interface (including the exact function needed in that place) and then simply replace with the generic interface with default BSL parameterization and optional FFTW for optimization. But I never contacted the other original author of that work regarding this issue yet.

**Comment by cosurgi** [comment]
> and then simply replace with the generic interface with default BSL parameterization and optional FFTW for optimization.

Yes. I guess this is our target goal here. I mentioned in one of other comments: if we want Boost.Math to not depend on Boost.Multiprecision and vice versa, then probably our work here should end up in a new library Boost.FFT.

EDIT: I believe the conclusion is to stay in [Boost.Math](https://github.com/BoostGSoC21/math/issues/16#issuecomment-894997648), at least for now. The careful required considerations for this are in the linked post.

**Comment by Lagrang3** [comment]
I did take a look at that file. There is no straightforward way to do the `FFTW_REDFT10` transform with our current interface. That one is a particular type of transform from the family of [the discrete cosine transforms](https://en.wikipedia.org/wiki/Discrete_cosine_transform) (DCT). We could probably express those DCTs as convolutions and then compute the convolutions using our interface, but that would not be as performant as a dedicated code to DCTs alone.

**Comment by ckormanyos** [comment]
> no straightforward way to do the `FFTW_REDFT10` transform with our current interface.

This means... post-GSoC21 in my world, unless it's simpler than it seems from your description.

Eduardo, you have accomplishe so much in this short time. I'm not sure if we should open a topic having such wide scope and depth on the final weekend. Kind of getting toward wrap-up/report time soon.

@Lagrang3 and @cosurgi
